app/utils/graph-area.js

```diff
@@ -0,0 +1,240 @@
+    tooltip,
+    stopped: false,
+  }
+
+  return {
+    start() {
+      start(params);
+    },
+    fit() {
+      fit({ el, x, margin, graph, clipPath, stripes });
+    },
+    destory() {
+      params.stopped = true;
+    }
+  };
+}
+
+function start(params) {
+  if (params.stopped) {
+    return
+  }
+
+  let all = [];
+  const series = [];
+  params.series.forEach(serie => {
+    const data = query(params.maxPoints, params.query, serie.field);
+    all = all.concat(data)
+    series.push({
+      field: serie.field,
+      lineChart: serie.lineChart,
+      areaChart: serie.areaChart,
+      data
+    })
+  })
+
+  updateTooltip(series, params)
+  updateAxis(all, params)
+  updateLines(series, params)
+}
+
+function fit(params) {
+  const margin = params.margin;
+  const width = getWeight(params.el, margin);
+
+  params.x.range([0, width]);
+  params.graph.attr('width', width + margin.left + margin.right);
+  params.clipPath.attr('width', width);
+  params.stripes.attr('width', width);
+}
+
+function query(maxPoints, query, field) {
+  let data = query(field.key) || []
+
+  if (data.length < maxPoints) {
+    const len = data.length
+    for (let i = 0; i < maxPoints - len; i++) {
+      data.unshift(null)
+    }
+  } else if (data.length > maxPoints) {
+    data = data.slice(-1 * maxPoints);
+  }
+
+  return data;
+}
+
+function adjustMax(dataMax, options) {
+  let optMinMax = options.minMax;
+  let optMax = options.max;
+  let optScaleDown = options.scaleDown;
+  let observedMax = options.observedMax;
+  let out = dataMax;
+
+  if (optMax) {
+    out = optMax;
+  } else if (optMinMax) {
+    out = Math.max(optMinMax, out);
+  }
+
+  if (observedMax && !optScaleDown) {
+    out = Math.max(observedMax, out);
+  }
+
+  if (!observedMax && out > 0 && options.maxDoubleInital) {
+    out *= 2;
+  }
+  options.observedMax = out;
+  return out;
+}
+
+function updateTooltip(series, params) {
+  params.tooltip.update(series)
+}
+
+function updateAxis(all, params) {
+  const min = params.min === null ? d3.min(all) : params.min;
+  const max = adjustMax(d3.max(all), params.options);
+
+  params.y.domain([min, max]);
+  params.y.range([params.height - 2, 2]);
+  params.y.rangeRound([params.height - 2, 2]);
+  params.yAxisG.call(params.yAxis);
+}
+
+function updateLines(series, params) {
+  series.forEach((serie, i) => {
+
+    serie.areaChart.attr('d', params.area(serie.data))
+      .attr('transform', null)
+      .transition()
+      .duration(params.duration)
+      .ease('linear')
+      .attr('transform', 'translate(' + params.x(-1) + ')')
+
+    serie.lineChart.attr('d', params.line(serie.data))
+      .attr('transform', null)
+      .transition()
+      .duration(params.duration)
+      .ease('linear')
+      .attr('transform', 'translate(' + params.x(-1) + ')')
+      .each('end', () => {
+        if (i === series.length - 1) {
+          start(params)
+        }
+      });
+
+  })
+}
+
+function drawStripes(svg, width, y, yTicks) {
```

Comment:
The memory graph (and only that one, possibly because the values are the smallest integers here?) has one more axis label than yTicks suggests, so the labels don't match up with the stripes:

![image](https://user-images.githubusercontent.com/753917/31695239-4136cbda-b35e-11e7-82d5-8ad42ab678ef.png)